File: Traverse a Tree/LevelOrderTraversal.java
import java.util.*;
import java.io.*;
import java.lang.*;

public class LevelOrderTraversal {

    public List<List<Integer>> levelOrder(TreeNode root) {

        List<List<Integer>> result = new ArrayList<>();

        if (root == null) {
            return result;
        }

        Deque<TreeNode> queue = new LinkedList<>();

        queue.addLast(root);

        while (!queue.isEmpty()) {

            int levelNum = queue.size();
            List<Integer> level = new ArrayList<>();

            for (int i = 0 ; i < levelNum; i++) {
                if (queue.peekFirst().left != null) {
                    queue.addLast(queue.peekFirst().left);
                }

                if (queue.peekFirst().right != null) {
                    queue.addLast(queue.peekFirst().right);
                }
                level.add(queue.pollFirst().val);
            }
            result.add(level);
        }
        return result;
    }

    public static void main(String[] args) throws IOException {

        //new LevelOrderTraversal().levelOrder(null);

        Traversal t = new Traversal();

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String line = br.readLine();

        System.out.println(new LevelOrderTraversal().levelOrder(t.stringToTreeNode(line)));

    }
}

File: Traverse a Tree/level_order_traversal.py
from collections import deque
from tree_node import TreeNode
from traversal import stringToTreeNode

import sys
import json


class LevelOrderTraversal:

    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """

        result = []

        if not root:
            return result

        queue = deque()
        queue.append(root)

        while len(queue) != 0:

            level = []
            nodeCount = len(queue)

            for _ in range(nodeCount):
                if len(queue) != 0 and queue[0].left:
                    queue.append(queue[0].left)
                if len(queue) != 0 and queue[0].right:
                    queue.append(queue[0].right)
                level.append(queue.popleft().val)

            result.append(level)

        return result


def readlines():
    for line in sys.stdin:
        yield line.strip('\n')


def stringToList(input):
    return json.loads(input)


if __name__ == '__main__':

    lines = readlines()
    line = next(lines)

    root = stringToTreeNode(line)

    print(LevelOrderTraversal().levelOrder(root))
